src/TropicalGeometry/groebner_basis.jl

```diff
@@ -0,0 +1,255 @@
+###
+# Computing (tropical) Groebner bases in Oscar
+# ============================================
+#
+# For a definition of tropical Groebner basis see Section 2.4 in:
+#   D. Maclagan, B. Sturmfels: Introduction to tropical geometry
+# To see how they can be computed using standard bases see:
+#   T. Markwig, Y. Ren: Computing tropical varieties over fields with valuation
+###
+
+
+
+#=======
+return true if f is homogeneous (w.r.t. total degree)
+return false otherwise
+=======#
+function is_homogeneous(f::Union{AbstractAlgebra.Generic.MPoly{K},fmpq_mpoly,fmpz_mpoly} where {K})
```

Comment:
The `where` is redundant, so you can write:
```suggestion
function is_homogeneous(f::Union{AbstractAlgebra.Generic.MPoly,fmpq_mpoly,fmpz_mpoly})
```
But actually... perhaps just take the common supertype?
```suggestion
function is_homogeneous(f::MPolyElem)
```